lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -284,12 +284,19 @@ public function diffTable(Table $table1, Table $table2)
             foreach ($toFkeys as $key2 => $constraint2) {
                 if ($this->diffForeignKey($constraint1, $constraint2) === false) {
                     unset($fromFkeys[$key1], $toFkeys[$key2]);
-                } else {
-                    if (strtolower($constraint1->getName()) === strtolower($constraint2->getName())) {
-                        $tableDifferences->changedForeignKeys[] = $constraint2;
-                        $changes++;
-                        unset($fromFkeys[$key1], $toFkeys[$key2]);
-                    }
+                    continue 2;
```

Comment:
The `continue 2;` are new, but i believe they were missing in the previous code. If the "fromKey" from the outer loop was matched to a "toKey", then we can skip to the next "fromKey", and don't need to match the current fromKey to another "toKey".

Reply:
This change in the code that is used by all platforms is only covered by an SQLite test. Would it be possible to rework the new test to use the DBAL APIs and run across platforms?

Reply:
Wouldn't applying https://github.com/doctrine/dbal/pull/4256/files#r484110298 be enough to do this? Or does "use the DBAL APIs" mean something else?

Reply:
@greg0ire, yes, it's exactly the same. I just reiterated because the previous comment wasn't taken into account.

Reply:
I tried it in https://github.com/doctrine/dbal/pull/4262 , and it works only for SQLite although I used a high level API.

Reply:
It is fine if it only works with SQLite right now, we can mark the test incomplete for other platforms. But this way, we at least identified the inconsistency in the API (which is a bug) which will be documented in the form of an incomplete test.